### PR TITLE
KOGITO-3090 - Fix pagination and add tests

### PR DESCRIPTION
**Pull request by r00ta**
Jira: https://issues.redhat.com/browse/KOGITO-3090

The pagination in the trusty service has some bugs, here a PR to implement it properly so that the contract with the UI is respected. Some tests are added.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/TrustyServiceImpl.java [hunk 1]
```diff
@@ -56,13 +61,23 @@ public TrustyServiceImpl(TrustyStorageService storageService) {
...
         Storage<String, Decision> storage = storageService.getDecisionsStorage();
         List<AttributeFilter<?>> filters = new ArrayList<>();
         filters.add(QueryFilterFactory.like(Execution.EXECUTION_ID_FIELD, prefix + "*"));
         filters.add(QueryFilterFactory.greaterThanEqual(Execution.EXECUTION_TIMESTAMP_FIELD, from.toInstant().toEpochMilli()));
         filters.add(QueryFilterFactory.lessThanEqual(Execution.EXECUTION_TIMESTAMP_FIELD, to.toInstant().toEpochMilli()));
-        return new ArrayList<>(storage.query().limit(limit).offset(offset).filter(filters).execute());
+        ArrayList result = new ArrayList<>(storage.query()
+                                                   .sort(asList(orderBy(Execution.EXECUTION_TIMESTAMP_FIELD, DESC)))
+                                                   .filter(filters)
+                                                   .execute()
```
Comment (danielezonca) [inline trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/TrustyServiceImpl.java:73]: Can you please better explain the original problem?
I don't like that now we are loading the entire storage in a list before applying the offset.

Comment (danielezonca) [reply trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/TrustyServiceImpl.java:73]: Ok I had a look to `Query` in `persistence-commons` and `count` is not supported so we should implement it to properly fix the bug

Comment (r00ta) [reply trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/TrustyServiceImpl.java:73]: Hi @danielezonca , 
we have two options: 
1) we load all the executions from that timestamp and then apply the offset/return the count of the results
2) we run two queries: one to extract the number of matched executions and one to extract only the results from `offset` to `offset+total`.

This PR implements the first option and adds some tests to check that the contract with the UI is not broken in terms of pagination. We can move forward with this and then create another ticket to implement the second option, or directly implement that. let me know what you think

**Comment by r00ta** [comment]
Hi @danielezonca , @kostola , @jiripetrlik , any update on this?

**Review by danielezonca — APPROVED** [review]
I think it is better to proceed with this PR and postpone to another ticket to refactor `Query` to support `count`. Can you please create a ticket for this?

**Comment by r00ta** [comment]
Jenkins retest this

**Review by jiripetrlik — APPROVED** [review]
Looks good to me.

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by r00ta** [comment]
> I think it is better to proceed with this PR and postpone to another ticket to refactor `Query` to support `count`. Can you please create a ticket for this?

https://issues.redhat.com/browse/KOGITO-3188

**Comment by r00ta** [comment]
jenkins retest this please

**Comment by r00ta** [comment]
jenkins retest this please

**Pull request merged**